x-pack/test/fleet_api_integration/apis/policy_secrets.ts

```diff
@@ -427,5 +584,67 @@ export default function (providerContext: FtrProviderContext) {
 
       expect(searchRes.hits.hits.length).to.eql(0);
     });
+
+    let fleetServerCheckPolicy: any;
+    it('should not store secrets if fleet server does not meet minimum version', async () => {
+      await createFleetServerAgent(fleetServerAgentPolicyId, 'server_1', '7.0.0');
+      await disableSecrets();
+      const { body: createResBody } = await createPolicyWithSecrets();
+
+      const createdPolicy = createResBody.item;
+
+      // secret should be in plain text i.e not a secret refrerence
+      expect(createdPolicy.vars.package_var_secret.value).eql('package_secret_val');
+
+      fleetServerCheckPolicy = createdPolicy;
+    });
+
+    it('should not store secrets if there are no fleet servers', async () => {
+      await clearAgents();
+
+      const { body: createResBody } = await createPolicyWithSecrets();
+
+      const createdPolicy = createResBody.item;
+
+      // secret should be in plain text i.e not a secret refrerence
+      expect(createdPolicy.vars.package_var_secret.value).eql('package_secret_val');
+    });
+
+    it('should convert plain text values to secrets once fleet server requirements are met', async () => {
+      if (!fleetServerCheckPolicy) {
+        throw new Error('fleetServerCheckPolicy not set, previous test must have failed');
```

Comment:
As a best practice, tests shouldn't rely on other tests, it makes it harder to debug one by one.